src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/ValidatorFactory.cs

```diff
@@ -42,10 +50,51 @@ public IValidator Create(Type validatorType)
                     $"{nameof(IDictionary<FeedType, SourceRepository>)} and a {loggerType.Name}!");
             }
 
-            return constructor.Invoke(new object[] 
+            /// Get the <see cref="ServiceIndexEntry"/> prefixes requested by this <see cref="IValidator"/>.
+            _indexHijacker.BeginGetUsedEntryPrefixes();
+            Construct(constructor);
+            var prefixes = _indexHijacker.EndGetUsedEntryPrefixes();
+
+            /// Iterate through all possible combinations of <see cref="ServiceIndexEntry"/>s that were requested.
+            var serviceIndexEntryEqualityComparer = new ServiceIndexEntryEqualityComparer();
+            var possibleCombinations = 
+                prefixes
+                    .Select(p => 
+                        new KeyValuePair<string, IEnumerable<ServiceIndexEntry>>(p, 
+                            _indexHijacker
+                                .GetServiceIndexEntriesByTypePrefix(p)
+                                    .Distinct(serviceIndexEntryEqualityComparer)));
+
+            IList<IValidator> validators = new List<IValidator>();
+
+            var combinations = CombinationUtils.Enumerate(possibleCombinations);
+            if (combinations != null && combinations.Count() > 0)
```

Comment:
CombinationUtils seems really complex. Could you simplify with SelectMany? Something like...

prefixes.SelectMany(
    p => _hijacker.GetServiceIndexEntriesByTypePrefix(p)
                 .Distinct()
                 .Select(e => new KeyValuePair(p, e) );